src/app.rs

```diff
@@ -96,43 +97,32 @@ impl AppData {
             channels.state.select(Some(0));
         }
 
-        let messages: Vec<_> = channels
-            .state
-            .selected()
-            .and_then(|idx| channels.items.get(idx))
-            .map(|channel| &channel.messages[..])
-            .unwrap_or(&[])
-            .iter()
-            .map(|msg| Message {
-                from: msg.from.clone(),
-                message: msg.message.clone(),
-                attachments: msg.attachments.clone(),
-                arrived_at: msg.arrived_at,
-            })
-            .collect();
-
-        let mut messages = StatefulList::with_items(messages);
-        if !messages.items.is_empty() {
-            messages.state.select(Some(0));
-        }
+        let chanpos = ChannelPosition {
+            top: 0,
+            upside: 0,
+            // value will be initialized in main.rs
+            downside: 0,
+        };
 
         Ok(AppData {
             channels,
-            messages,
+            chanpos,
             input: String::new(),
             input_cursor: 0,
             input_cursor_chars: 0,
         })
     }
 }
 
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Derivative, Serialize, Deserialize)]
+#[derivative(Debug)]
 pub struct Channel {
     /// Either phone number or group id
     pub id: String,
     pub name: String,
     pub is_group: bool,
-    pub messages: Vec<Message>,
+    #[derivative(Debug = "ignore")]
```

Comment:
Nice! Did not know about this crate.

Reply:
I read up on how to rebase and squash commits and cleaned things up as best I could. Thanks for the suggestion and offer to help. Let me know if you had anything else in mind for this.